machine/v1beta1/types_vsphereprovider.go

```diff
@@ -198,8 +198,30 @@ type VSphereDisk struct {
 	// +kubebuilder:validation:Maximum=16384
 	// +required
 	SizeGiB int32 `json:"sizeGiB"`
+	// provisioningMode specifies the provisioning type to be used by this vSphere data disk.
+	// If not set, the setting will be provided by the default storage policy.
+	// +optional
```

Comment:
Include an explicit mention in the Godoc that the field is optional. Users using tools like `oc explain ...` are not able to see the markers we use so an explicit mention is helpful for users to know which fields are optional or required

Reply:
I'll update this.